pkg/process/procutil/process_linux.go

```diff
@@ -226,27 +242,20 @@ func (p *Probe) ProcessesByPID(now time.Time) (map[int32]*Process, error) {
 			Cwd:     p.getLinkWithAuthCheck(pathForPID, "cwd"), // /proc/[pid]/cwd, requires permission checks
 			Exe:     p.getLinkWithAuthCheck(pathForPID, "exe"), // /proc/[pid]/exe, requires permission checks
 			NsPid:   statusInfo.nspid,                          // /proc/[pid]/status
-			Stats: &Stats{
-				CreateTime:  statInfo.createTime,    // /proc/[pid]/stat
-				Status:      statusInfo.status,      // /proc/[pid]/status
-				Nice:        statInfo.nice,          // /proc/[pid]/stat
-				CPUTime:     statInfo.cpuStat,       // /proc/[pid]/stat
-				MemInfo:     statusInfo.memInfo,     // /proc/[pid]/status
-				MemInfoEx:   memInfoEx,              // /proc/[pid]/statm
-				CtxSwitches: statusInfo.ctxSwitches, // /proc/[pid]/status
-				NumThreads:  statusInfo.numThreads,  // /proc/[pid]/status
-			},
-		}
-		if p.withPermission {
-			proc.Stats.OpenFdCount = p.getFDCountImproved(pathForPID) // /proc/[pid]/fd, requires permission checks
-			proc.Stats.IOStat = p.parseIO(pathForPID)                 // /proc/[pid]/io, requires permission checks
-		} else {
-			proc.Stats.IOStat = &IOCountersStat{
-				ReadCount:  -1,
-				WriteCount: -1,
-				ReadBytes:  -1,
-				WriteBytes: -1,
-			} // use -1 values to represent "no permission"
+			Stats:   stats,
+		}
+		if collectStats {
```

Comment:
why not use `if collectStats && p.withPermission` to avoid creating another nested if block?

Reply:
The -1 definitely struck me as an odd design choice, especially when it came down to testing. It was like this before however, so we need to be careful with changing this

Reply:
Gotcha, I went ahead and changed it to `-1` :)